spark/v3.3/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -643,10 +642,9 @@ public void testSingleCommitWithRewriteFailure() {
         .when(spyRewrite)
         .rewriteFiles(any(), argThat(failGroup));
 
-    AssertHelpers.assertThrows(
-        "Should fail entire rewrite if part fails",
-        RuntimeException.class,
-        () -> spyRewrite.execute());
+    Assertions.assertThatThrownBy(() -> spyRewrite.execute())
+        .as("Should fail entire rewrite if part fails")
+        .isInstanceOf(RuntimeException.class);
```

Comment:
all of these should have a `hasMessage()` check. You can use the Spark 3.5 version of that test class as a reference

Reply:
I agree that we should always assert the message and this it is bad that this code was not doing that before the changes. do you view the change as a reduction in overall quality and therefore a blocker for the PR?

Reply:
in all previous PRs where we moved from AssertHelpers to AssertJ we also improved cases like this, so we should do the same in this class

Reply:
done! added `Assert on exception messages where converted from AssertHelpers` commit, PTAL